types/src/crypto/asymmetric_key.rs

```diff
@@ -314,8 +313,8 @@ pub enum PublicKey {
     #[data_size(skip)] // Manually verified to have no data on the heap.
     Ed25519(ed25519_dalek::PublicKey),
     /// secp256k1 public key.
-    #[data_size(skip)] // Manually verified to have no data on the heap.
-    Secp256k1(k256::PublicKey),
+    #[data_size(skip)]
+    Secp256k1([u8; SECP256K1_COMPRESSED_PUBLIC_KEY_LENGTH]),
```

Comment:
There's no full-size variant. Is that intentional?

Reply:
Yes - we didn't want to deal with a mixture of uncompressed and compressed keys.  Our constructors only allow us to hold compressed keys.

Reply:
Yes, we must have a canonical representation for public keys.

Reply:
> Yes - we didn't want to deal with a mixture of uncompressed and compressed keys. Our constructors only allow us to hold compressed keys.

Probably worth mentioning in the docs for client integration.